File: juggleim-rnsdk/ios/GenericCustomMessage.m
#import "GenericCustomMessage.h"

static NSString *jsType = @"jgrn:custom";

@implementation GenericCustomMessage

+ (void)setJsType:(NSString *)type {
  jsType = type;
}

+ (NSString *)contentType {
  return jsType;
}

- (NSData *)encode {
  return self.data;
}

- (void)decode:(NSData *)data {
  self.data = data;
}

- (void)setData:(NSData *)data {
  _data = data;
}

- (NSData *)getData {
  return _data;
}

- (NSString *)conversationDigest {
  return @"[自定义消息]";
}

@end

File: juggleim-rnsdk/android/src/main/java/com/juggleim/GenericCustomMessage.java
package com.juggleim;

import com.juggle.im.model.MessageContent;
import java.nio.charset.StandardCharsets;

/**
 * 通用自定义消息类
 * 用于承载 RN 层自定义消息的 JSON 数据
 */
public class GenericCustomMessage extends MessageContent {
    private String mContentType;
    private byte[] mData;

    private static String jsType = "jgrn:custom";
    
    public GenericCustomMessage() {
        this.mContentType = jsType;
    }

    public static void setJsType(String type) {
        jsType = type;
    }
    
    @Override
    public String getContentType() {
        return mContentType;
    }
    
    @Override
    public byte[] encode() {
        return mData;
    }
    
    @Override
    public void decode(byte[] data) {
        this.mData = data;
    }
    
    public void setData(byte[] data) {
        this.mData = data;
    }
    
    public byte[] getData() {
        return mData;
    }
    
    @Override
    public String conversationDigest() {
        // 返回默认摘要
        return "[自定义消息]";
    }
}
